File: sep_04/Fibonacci.java
// 2. WAP to find the Fibonacci series upto N terms using recursion.

import java.util.Scanner;

public class Fibonacci {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);

    System.out.print("Enter the no. of terms: ");
    int terms = sc.nextInt();
    sc.close();

    int sum = fibonacci(terms);
    System.out.printf("Fibonacci series upto %d terms:\n", terms);
  }

  public static int fibonacci(int upto) {
    if (upto < 1) return 0;
    return upto + fibonacci(upto - 1);
  }
}

File: nov_07/fibonacci.py
def fibonacci(max_terms, seperator):
  first = 0
  second = 0

  for i in range(max_terms):
    sum = first + second
    print(first, end=seperator)
    first = second + 1
    second = sum

n_terms = int(input("Enter the number of terms to print the fibonacci series: "))

print("Fibonacci series upto", n_terms, "terms:")
fibonacci(n_terms, " ")
